### PR TITLE
add library for parrllel 

### PR DESCRIPTION
**Pull request by zixunlu**

### CODE REVIEW

**Comment by zixunlu** [comment]
I'm writing simple function by using mpi and combine with our function as soon as possible.

File: setup_parallel.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+apt install mpich2
```
Comment (pop) [inline setup_parallel.sh]: Please move this to `vagrant/setup.sh` along with all of the other package installations.

Comment (zixunlu) [reply setup_parallel.sh]: ok

Comment (pop) [reply setup_parallel.sh]: ... also remove this file once you added it to vagrant/setup.sh...

**Pull request closed without merging**